htdocs/core/class/commonobject.class.php

```diff
@@ -4181,16 +4181,25 @@ public function getRights()
 	 * @param int 		$origin_id 		Old thirdparty id (the thirdparty to delete)
 	 * @param int 		$dest_id 		New thirdparty id (the thirdparty that will received element of the other)
 	 * @param array 	$tables 		Tables that need to be changed
+	 * @param bool		$delete_if_error Delete if udpate failed because of duplicate entry (on category_societe for example)
 	 * @return bool
 	 */
-	public static function commonReplaceThirdparty(DoliDB $db, $origin_id, $dest_id, array $tables)
+	public static function commonReplaceThirdparty(DoliDB $db, $origin_id, $dest_id, array $tables, $delete_if_error = false)
 	{
 		foreach ($tables as $table)
 		{
 			$sql = 'UPDATE '.MAIN_DB_PREFIX.$table.' SET fk_soc = '.$dest_id.' WHERE fk_soc = '.$origin_id;
 
 			if (!$db->query($sql)) {
-				return false;
+				if($delete_if_error) {
```

Comment:
This looks not enough. If there is categ A-B on kept thirdarty and B-C on old one, we must get A-B-C after merge. Not A-B. A special test on table == llx_categories seems required to include a DELETE of category on second thirparty that is already on first one before running the update is required.